Libplanet/Net/TxCompletion.cs

```diff
@@ -149,31 +150,42 @@ private HashSet<TxId> GetRequiredTxIds(IEnumerable<TxId> ids)
                                 return false;
                             }));
 
+                    var validTxs = new List<Transaction<TAction>>();
+                    IImmutableSet<TxId> stagedTxIds = _blockChain.GetStagedTransactionIds();
                     foreach (var tx in txs)
                     {
                         try
                         {
-                            _blockChain.StageTransaction(tx);
+                            if (!stagedTxIds.Contains(tx.Id))
+                            {
+                                _blockChain.StageTransaction(tx);
+                                validTxs.Add(tx);
+                            }
                         }
                         catch (InvalidTxException ite)
                         {
                             _logger.Error(
                                 ite,
-                                "Transaction {TxId} will not be staged since it is invalid.",
-                                tx.Id);
+                                "Transaction {TxId} will not be staged since it is invalid. {E}",
+                                tx.Id,
+                                ite);
```

Comment:
It might be better not to pass `ite` here.  See also <https://github.com/planetarium/libplanet/issues/1632>.